src/components/TermSelector/index.js

```diff
@@ -0,0 +1,80 @@
+import React, { ReactNode } from 'react';
+
+import { FormTokenField } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Generate a list of term IDs, keyed by title.
+ *
+ * @param {object[]} termObjects array of terms, returned from getEntityRecords.
+ * @returns {Array} array of term IDs, keyed by title.
+ */
+function generateTitleToIdMap( termObjects ) {
+
+	if ( ! termObjects ) {
+		return [];
+	}
+
+	return termObjects.reduce( ( accumulator, currentTerm ) => {
+		accumulator[currentTerm.name] = currentTerm.id;
+		return accumulator;
+	}, {} );
+
+}
+
+/**
+ * Generate a list of term titles, keyed by id.
+ *
+ * @param {object[]} termObjects array of terms, returned from getEntityRecords.
+ * @returns {Array} array of term titles, keyed by id.
+ */
+function generateIdToTitleMap( termObjects ) {
+
+	if ( ! termObjects ) {
+		return [];
+	}
+
+	return termObjects.reduce( ( accumulator, currentTerm ) => {
+		accumulator[currentTerm.id] = currentTerm.name;
+		return accumulator;
+	}, {} );
+}
+
+/**
+ * Form token field to select one or more terms.
+ *
+ * @param {object} props Component props
+ * @returns {ReactNode} Component
+ */
+function TermSelector( props ) {
+	const { taxonomy, value = [], onChange } = props;
+
+	const taxObject = useSelect( select => {
+		return select( 'core' ).getTaxonomy( taxonomy );
+	}, [ taxonomy ] );
+
+	const { taxonomyTermsById, taxonomyTermsByTitle } = useSelect( ( select ) => {
+		const termObjects = select( 'core' ).getEntityRecords( 'taxonomy', taxonomy, { per_page: 100 } ) ?? [];
+		const taxonomyTermsById = generateIdToTitleMap( termObjects );
+		const taxonomyTermsByTitle = generateTitleToIdMap( termObjects );
+
+		return {
+			taxonomyTermsById,
+			taxonomyTermsByTitle,
+		};
+	}, [ taxonomy ] );
```

Comment:
Been thinking about this functionality a bit and I actually don't think this is ideal for 2 reasons. 

1. This is potentially a large and slow request on initial load. It's very easy to create a PostPicker with filters several taxonomies too. 
2. It's still limited to 100. This covers most smaller sites fine... but will be a limitation pretty quickly on larger sites. 

I think the old Term Select UI in HM Gutenberg Tools handled this using React Select library, that showed ~10 by default but supported pagination and searching too. 

Not saying we should re-implement this exactly... but given this field is visually a text input, I think it should make a request as you type to get results matching your query.

Reply:
This is a very valid point, the initial use case I created this for didn't come up against these limits but good to future proof it.

Reply:
how does core solve this? I know that some REST API requests are prefetched and bundled with the page, and I'd expect that to be the case for some terms in taxonomies that match the currently edited post. Also would this not trigger when the modal is opened rather than when the page is loaded?

If so, is this really a blocker? Or a performance enhancement in a small update?